apps/wallet/src/ui/app/pages/home/nfts/index.tsx

```diff
@@ -1,47 +1,54 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { Coin } from '@mysten/sui.js';
+import { useMemo } from 'react';
 import { Link } from 'react-router-dom';
 
 import Alert from '_components/alert';
 import { ErrorBoundary } from '_components/error-boundary';
 import Loading from '_components/loading';
-import NFTdisplay from '_components/nft-display';
-import { useAppSelector, useObjectsState } from '_hooks';
-import { accountNftsSelector } from '_redux/slices/account';
+import { NFTDisplayCard } from '_components/nft-display';
+import { useAppSelector, useObjectsOwnedByAddress } from '_hooks';
 import PageTitle from '_src/ui/app/shared/PageTitle';
 
 function NftsPage() {
-    const nfts = useAppSelector(accountNftsSelector);
-    const { error, loading, showError } = useObjectsState();
-    const isNftsFound = nfts.length > 0;
+    const accountAddress = useAppSelector(({ account }) => account.address);
+    const { data, isLoading, error, isError } =
+        useObjectsOwnedByAddress(accountAddress);
+    const nfts = useMemo(
+        () => data?.filter((obj) => !Coin.isCoin(obj)),
+        [data]
+    );
+
     return (
         <div className="flex flex-col flex-nowrap items-center gap-4 flex-1">
             <PageTitle title="NFTs" />
-            <Loading loading={loading}>
-                {showError && error ? (
+            <Loading loading={isLoading}>
+                {isError ? (
                     <Alert>
                         <div>
                             <strong>Sync error (data might be outdated)</strong>
                         </div>
-                        <small>{error.message}</small>
+                        <small>{(error as Error).message}</small>
                     </Alert>
                 ) : null}
-                {isNftsFound ? (
-                    <div className="grid grid-cols-2 gap-x-3.5 gap-y-4">
-                        {nfts.map((nft) => (
+
+                {nfts ? (
```

Comment:
`nfts` can be an empty array. We should show the no nfts message in that case as well